notify/webex/webex.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2022 Prometheus Team
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package webex
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	commoncfg "github.com/prometheus/common/config"
+
+	"github.com/prometheus/alertmanager/config"
+	"github.com/prometheus/alertmanager/notify"
+	"github.com/prometheus/alertmanager/template"
+	"github.com/prometheus/alertmanager/types"
+)
+
+const (
+	// maxMessageSize represents the maximum message length that Webex supports.
+	maxMessageSize = 7439
+)
+
+type Notifier struct {
+	conf    *config.WebexConfig
+	tmpl    *template.Template
+	logger  log.Logger
+	client  *http.Client
+	retrier *notify.Retrier
+	APIURL  *config.URL
+}
+
+// New returns a new Webex notifier.
+func New(c *config.WebexConfig, t *template.Template, l log.Logger, httpOpts ...commoncfg.HTTPClientOption) (*Notifier, error) {
+	if c.HTTPConfig.Authorization == nil && c.HTTPConfig.BearerToken == "" {
+		c.HTTPConfig.Authorization = &commoncfg.Authorization{
+			Type:        "Bearer",
+			Credentials: commoncfg.Secret(c.BotToken),
+		}
+	}
+
+	client, err := commoncfg.NewClientFromConfig(*c.HTTPConfig, "webex", httpOpts...)
+	if err != nil {
+		return nil, err
+	}
+
+	n := &Notifier{
+		conf:    c,
+		tmpl:    t,
+		logger:  l,
+		client:  client,
+		retrier: &notify.Retrier{},
+		APIURL:  c.APIURL,
+	}
+
+	return n, nil
+}
+
+type webhook struct {
+	Message string `json:"markdown"`
```

Comment:
It took me a few minutes to see that the field's name didn't match the field's tag and I was confused that the [Webex messages API](https://developer.webex.com/docs/api/v1/messages/create-a-message) had no `message` parameter.

```suggestion
	Markdown string `json:"markdown"`
```

Reply:
Done!